Update User Guide, Developer Guide v1.3 (#75)
AY2021S1-CS2103T-T09-1/tp: docs/DeveloperGuide.md

(For all use cases below, the **System** is the `GradPad` and the **Actor** is the `user`, unless specified otherwise)

**Use case: UC01 Delete a Module from `Current Modules`**

**MSS**

1.  User requests to delete a specific Module in the `Current Modules`
2.  GradPad deletes the module

> 3.  GradPad displayed the deleted module onto the `Command Line Display`

Reviewer: ```suggestion
3.  GradPad displays the deleted module onto the `Command Line Display`
```

Author: Ah grammar mistakes

    Use case ends.
    
**Use case : UC04 View all current modules**

**MSS**

1. User requests to view list of modules in `Current Modules`
2. GradPad shows all modules added into `Current Modules`

    Use case ends.
    

> **User case: UC05 Edit a Module in `Current Modules`**

Reviewer: ```suggestion
**Use case: UC05 Edit a Module in `Current Modules`**
```

1. User requests to list all modules in `Current Modules`
2. GradPad shows the list of modules in `Current Modules`

> 3. User requests to edit a module in `Current Modules`

Reviewer: I think we also need an extension to cover the case when invalid fields are given e.g. `credit/5` instead of `cr/5`

4. Module is replaced with updated fields
    
    Use case ends.

**Extensions**

* 2a. The list of modules in `Current Modules` is empty.

    Use case ends.
    
* 3a. The given index is invalid.
    
    * 3a1. GradPad shows an error message.
    
        Use case resumes at step 2.

**Use case : UC06 View required modules in syllabus**

**MSS**

1. User requests to view all required modules in syllabus
2. GradPad displays the required modules in syllabus onto the `Command Line Display`

    Use case ends.
  
**Use case : UC07 Search for module details**

**MSS**

> 1. User requests to search a module in the required module list.

Reviewer: Actually it's not just the required module list righttt cos that's only like ~10 mods out of all 150 CS mods that they can search for

```suggestion
1. User requests to search for a module in the CS curriculum.
```
